internal/api/proxy.go

```diff
@@ -0,0 +1,138 @@
+package api
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+)
+
+func applyProxy(transport *http.Transport, proxyURL *url.URL, proxyPath string) (applied bool) {
+	if proxyURL == nil && proxyPath == "" {
+		return false
+	}
+
+	handshakeTLS := func(ctx context.Context, conn net.Conn, addr string) (net.Conn, error) {
+		// Extract the hostname (without the port) for TLS SNI
+		host, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			return nil, err
+		}
+		// Be sure to clone TLS-specific config settings from the transport,
+		// like InsecureSkipVerify.
+		tlsConn := tls.Client(conn, &tls.Config{
+			ServerName:         host,
+			InsecureSkipVerify: transport.TLSClientConfig.InsecureSkipVerify,
+		})
```

Comment:
Are there any other TLS configs we need to copy? Woudl it be safer to do something like this?

```suggestion
		tlsConfig := *transport.TLSClientConfig
		tlsConfig.ServerName = host
		tlsConn := tls.Client(conn, &tlsConfig)
```

Reply:
I don't know that we really want to share things like client certs and session tickets. Let's stick with `InsecureSkipVerify` for now and I'll update the comment.